app/src/main/java/com/example/eventscan/Helpers/GeolocationHandler.java

```diff
@@ -0,0 +1,110 @@
+package com.example.eventscan.Helpers;
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationManager;
+import android.location.LocationListener;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+public class GeolocationHandler {
+
+    private static boolean locationUpdatesEnabled = false;
+    private static LocationManager locationManager;
+    private static LocationListener locationListener;
+    private static final int LOCATION_PERMISSION_REQUEST_CODE = 1001;
+
+    public static void enableLocationUpdates(Context context) {
+        // Initialize location manager and listener if not already initialized
+        if (locationManager == null) {
+            locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+            locationListener = createLocationListener();
+        }
+
+        // Check for location permissions
+        if (checkLocationPermissions(context)) {
+            // Location permissions granted, enable updates
+            toggleLocationUpdates();
```

Comment:
if enableLocationUpdates() is called twice in a row, will the location be disabled due to the toggle call?

Reply:
Fixed. Thank you for catching that